src/index.js

```diff
@@ -1,3 +1,141 @@
+ * @module Multibase
+ */
 'use strict'
 
-exports = module.exports = require('./multibase')
+const constants = require('./constants')
+
+exports = module.exports = multibase
+exports.encode = encode
+exports.decode = decode
+exports.isEncoded = isEncoded
+
+const errNotSupported = new Error('Unsupported encoding')
+
+/**
+ * Create a new buffer with the multibase varint+code.
+ *
+ * @param {string|number} nameOrCode - The multibase name or code number.
+ * @param {Buffer} buf - The data to be prefixed with multibase.
+ * @memberof Multibase
+ * @returns {Buffer}
+ */
+function multibase (nameOrCode, buf) {
+  if (!buf) {
+    throw new Error('requires an encoded buffer')
+  }
+  const base = getBase(nameOrCode)
+  const codeBuf = new Buffer(base.code)
+
+  const name = base.name
+  validEncode(name, buf)
+  return Buffer.concat([codeBuf, buf])
+}
+
+/**
+ * Encode data with the specified base and add the multibase prefix.
+ *
+ * @param {string|number} nameOrCode - The multibase name or code number.
+ * @param {Buffer} buf - The data to be encoded.
+ * @returns {Buffer}
+ * @memberof Multibase
+ */
+function encode (nameOrCode, buf) {
+  const base = getBase(nameOrCode)
+  const name = base.name
+
+  return multibase(name, new Buffer(base.encode(buf)))
+}
+
+/**
+ *
+ * Takes a buffer or string encoded with multibase header
+ * decodes it and returns an object with the decoded buffer
+ * and the encoded type { base: <name>, data: <buffer> }
+ *
+ * from @theobat : This is not what the multibase.spec.js test is waiting for,
+ * hence the return decodeObject.data
+ *
+ * @param {Buffer|string} bufOrString
+ * @returns {Object} result
+ * @returns {string} result.base
+ * @returns {Buffer} result.data
+ * @memberof Multibase
+ *
+ */
+function decode (bufOrString) {
+  if (Buffer.isBuffer(bufOrString)) {
+    bufOrString = bufOrString.toString()
+  }
+
+  const code = bufOrString.substring(0, 1)
+  bufOrString = bufOrString.substring(1, bufOrString.length)
+
+  if (typeof bufOrString === 'string') {
+    bufOrString = new Buffer(bufOrString)
+  }
+
+  const base = getBase(code)
+
+  const decodeObject = {
+    base: base.name,
+    data: new Buffer(base.decode(bufOrString.toString()))
+  }
+  return decodeObject.data
+}
+
+/**
+ * Is the given data multibase encoded?
+ *
+ * @param {Buffer|string} bufOrString
+ * @returns {boolean}
+ * @memberof Multibase
+ */
+function isEncoded (bufOrString) {
+  if (Buffer.isBuffer(bufOrString)) {
+    bufOrString = bufOrString.toString()
+  }
+
+  const code = bufOrString.substring(0, 1)
+  try {
+    const base = getBase(code)
+    return base.name
+  } catch (err) {
+    return false
+  }
+}
+
+/**
+ * @param {string} name
+ * @param {Buffer} buf
+ * @private
+ * @returns {undefined}
+ */
+function validEncode (name, buf) {
+  const base = getBase(name)
+  base.decode(buf.toString())
+}
+
+/**
+ * @param {string|number} nameOrCode
+ * @returns {Base}
```

Comment:
Hm, so people get the idea that a `Base` is returned, but they have no idea of what a base is. We should document at least what the base includes here, to not confuse people.